### DISCUSSION TITLE
Overwrite CRA css styles in "layer components"

### DISCUSSION THREAD

**Discussion by talgat-ruby** [tailwindlabs/tailwindcss#3700] [Help] [answered]
I bootstrapped with create-react-app. Followed instructions to install tailwind. However cannot overwrite styles applied in `components` layer. Please help.

```
// index.css

@tailwind base;
@tailwind components;
@tailwind utilities;
```

Somewhere in my components I have
```
import "./AllDataSearch.css";

<Search className="fu mr-4 w-6 h-6 fill-gray-4" />
```

And css file 
```
// AllDataSearch.css

@layer components {
  .fu {
    @apply mr-6 fill-green;
  }
}
```

And my output is
<img width="375" alt="Screen Shot 2021-03-06 at 13 02 25" src="https://user-images.githubusercontent.com/17277013/110220717-5dd1b100-7e7c-11eb-8e62-dc809e12a0cf.png">
As you can see `.fu` actually has higher priority.

Is it expected behavior? My expectation was that utility should override the components layer.
If not I would love any suggestions or comments.

**Comment by adamwathan** [comment, marked as answer]
Can you share an actual reproduction repo? My suspicion is your layer code is in a separate file that you are importing in JS instead of in CSS which is causing it to be processed totally independently so Tailwind is running twice and has no ability to combine the styles. Make sure you are doing all your separate file CSS stuff in a single CSS tree by combining them all via CSS imports in one main CSS file and only importing that one file in JS 👍🏻

**Reply by talgat-ruby** [reply]
@adamwathan Thank you. I couldn't create my exact configuration in sandbox. But you are right I used [CRA stylesheet suggestion](https://create-react-app.dev/docs/adding-a-stylesheet). And when I tried to import to a single `index.css` it worked. I would love to add it to docs if you can give tips how to do that. Also worth mentioning is there a way I can use css modules with this approach? So less headache on unique naming components classes.

**Reply by MatthiasDunker** [reply]
Hi the both of you!
I have the same “problem“.
Especially in the JS ui framework world referencing all CSS in one file instead of keeping component code next to each other is very unfamiliar. If I understand it correctly, I cannot create components interoperably, but always have to do the CSS import manually in the projects global CSS file.
Is it meant this way?
I already tried to use [twin.macro](https://github.com/ben-rogerson/twin.macro), but it does not recognize the `@layer` directive.

**Reply by meotimdihia** [reply]
I have the same problem, I can't overwrite in 
```css
@layer components/base {
  select {
    @apply bg-gray-100 dark:bg-gray-700 
  }
}
```

in my react website:
```jsx
// not work
<select className="bg-gray-800" /> 
```

```jsx
// work
<select className="!bg-gray-800" /> 
```
